projects/libs/core-library-typescript/test/consensus/block-builder.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { types } from "../../src/common-library/types";
+import * as chai from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import * as sinonChai from "sinon-chai";
+import { stubInterface } from "ts-sinon";
+import BlockBuilder from "../../src/consensus/block-builder";
+
+chai.should();
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+const transactionPool = stubInterface<types.TransactionPoolClient>();
+const virtualMachine = stubInterface<types.VirtualMachineClient>();
+
+function randomUInt() {
+  return Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+}
+
+function aRandomTransaction(): types.Transaction {
+    return {
+        version: 1,
+        sender: randomUInt().toString(16),
+        contractAddress:  randomUInt().toString(16),
+        payload: Math.random().toString(),
+        signature: ""
+    };
+}
+
+function aDummyTransactionSet(numberOfTransactions = 3) {
+  const transactions = [];
+  for (let i = 0; i < numberOfTransactions; i++); {
+    transactions.push(aRandomTransaction);
+  }
+
+  return transactions;
+}
+
+function aRandomStateDiff(): types.modifiedStateKey[] {
+  return [
+    {
+      contractAddress: randomUInt().toString(16),
```

Comment:
Generating random numbers in tests is considered a bad practice, as long as there is no way to use a seed and artifact it, so that it'd be possible to replay the tests again and again (in order to reproduce/fix the issue). This is also true when randomizing test order, timeouts/delays, etc.

In this case, could you just generate once a pool of numbers and test it as a table test? Alternatively, we can add having a `process.env.TEST_SEED` and make sure that it's exported or logged, but I personally think that the former would be much simpler and cleaner at this point.